event_track_generate/models.py

```diff
@@ -0,0 +1,194 @@
+        default=lambda self:
+            self.env["event.event"].browse(self.env.context["active_id"]),
+        required=True)
+    event_date_begin = fields.Datetime(
+        related="event_id.date_begin",
+        help="Change it in the event form. "
+             "No tracks before this date will be generated.")
+    event_date_end = fields.Datetime(
+        related="event_id.date_end",
+        help="Change it in the event form. "
+             "No tracks after this date will be generated.")
+    event_date_tz = fields.Selection(
+        related="event_id.date_tz",
+        help="Change it in the event form. Timezone of the generated tracks.")
+    name = fields.Char(
+        "Track title",
+        required=True,
+        help="Title that will be assigned to all created tracks.")
+    location_id = fields.Many2one("event.track.location", "Location")
+    speaker_ids = fields.Many2many(
+        "res.partner",
+        string="Speakers",
+        relation="event_track_generate_generator_speaker_ids")
+    tag_ids = fields.Many2many(
+        "event.track.tag",
+        string="Tags",
+        relation="event_track_generate_generator_tag_ids")
+    mondays = fields.Boolean(help="Create tracks on Mondays.")
+    tuesdays = fields.Boolean(help="Create tracks on Tuesdays.")
+    wednesdays = fields.Boolean(help="Create tracks on Wednesdays.")
+    thursdays = fields.Boolean(help="Create tracks on Thursdays.")
+    fridays = fields.Boolean(help="Create tracks on Fridays.")
+    saturdays = fields.Boolean(help="Create tracks on Saturdays.")
+    sundays = fields.Boolean(help="Create tracks on Sundays.")
+    start_time = fields.Float(
+        required=True,
+        help="Each track will start at this time (in the event's timezone).")
+    duration = fields.Float(
+        required=True,
+        help="Each track will have this duration.")
+    end_time = fields.Float(
+        compute="_compute_end_time",
+        help="Each track will end at this time.")
+    delete_existing_tracks = fields.Boolean()
+    publish_tracks_in_website = fields.Boolean()
+    adjust_start_time = fields.Boolean(
+        default=True,
+        help="Make event's start time match the start of the first track.")
+    adjust_end_time = fields.Boolean(
+        default=True,
+        help="Make event's end time match the end of the last track.")
+
+    @api.one
+    @api.depends("start_time", "duration")
+    def _compute_end_time(self):
+        self.end_time = self.start_time + self.duration
+
+    @api.one
+    def action_generate(self):
+        """Generate event tracks according to received data.
+
+        This is the main method of this class, triggered by the UI.
+        """
+        # You need at least one weekday
+        weekdays = self.weekdays()[0]
+        if True not in weekdays:
+            raise exceptions.NoWeekdaysError()
+
+        # Delete existing
+        if self.delete_existing_tracks:
+            self.event_id.track_ids.exists().unlink()
```

Comment:
Remove `exists()`. The unlink is done the same way, independently if there are records or not